align/pnr/checker.py

```diff
@@ -0,0 +1,26 @@
+from ..schema import constraint
+from ..cell_fabric import transformation
+
+
+def check_placement(placement_verilog_d):
+    for module in placement_verilog_d['modules']:
+        if 'constraints' not in module or len(module['constraints']) == 0:
+            continue  # No constraints
+        constraints = constraint.ConstraintDB(module['constraints'])
+        if sum(hasattr(x, 'check') for x in constraints) == 0:
+            continue  # Nothing useful to check against
+        for inst in module['instances']:
+            t = inst['transformation']
+            r = next((x['bbox'] for x in placement_verilog_d['modules'] if x['name'] == inst['template_name']),
+                     next((x['bbox'] for x in placement_verilog_d['leaves'] if x['name'] == inst['template_name']), None))
+            assert r is not None, f'Could not find {inst["template_name"]} in modules or leaves!'
+            bbox = transformation.Transformation(**t).hitRect(transformation.Rect(*r)).canonical()
```

Comment:
Good use of the Transformation and Rect methods. That kwargs trick is pretty cool, isn't it?